src/Simulator/IncrementalRedstoneSimulator/RedstoneWireHandler.h

```diff
@@ -14,12 +14,34 @@ class cRedstoneWireHandler:
 
 public:
 
-	inline static bool IsDirectlyConnectingMechanism(BLOCKTYPE a_Block)
+	inline static bool IsDirectlyConnectingMechanism(cWorld & a_World, BLOCKTYPE a_Block, Vector3i a_QueryPosition, Vector3i a_Offset)
 	{
 		switch (a_Block)
 		{
 			case E_BLOCK_REDSTONE_REPEATER_ON:
 			case E_BLOCK_REDSTONE_REPEATER_OFF:
+				if ((a_World.GetBlockMeta(a_QueryPosition) & 0x1) == 1)
```

Comment:
This line needs its magic numbers replacing with either named constants or, better, a function that already exists with semantic meaning. Some comments for what the two cases mean are also necessary.